### PR TITLE
ENH: Force modern python3 find package and install of .pyi files

### PR DESCRIPTION
**Pull request by hjmjohnson**
This PR represents the rabbit hole untangling of finding Python with a consistent version across internal remote module builds with python wrapping turned on.

The end goal was to test installing of .pyi files in the itk-stub directory for the purpose of creating wheels that provide type hinting to the IDE.

## PR Checklist
- [X] No API changes(https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- UPDATED MIN PYTHON VERSION CHECKING TO 3.7
- [X] No major design changes(https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

**Review by dzenanz — COMMENTED** [review]
Some in-line comments.

File: Utilities/Maintenance/FindPython3_ModernizeCMake.sh [hunk 1]
```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+# \author Hans J. Johnson
```
[the remark this answers is not in the document]
Comment (hjmjohnson) [reply Utilities/Maintenance/FindPython3_ModernizeCMake.sh]: DONE.

File: CMake/WrappingConfigCommon.cmake [hunk 2]
```diff
@@ -0,0 +1,46 @@
+
```
[the remark this answers is not in the document]
Comment (hjmjohnson) [reply CMake/WrappingConfigCommon.cmake]: Done.

**Comment by hjmjohnson** [comment]
@dzenanz This is ready to review and rebase merge if the ghostflow-check-master can be overlooked until it is fixed.

This PR includes #3561 and #3562 in order to get this to be a clean build.

@thewtex This cleans up the python building environment, and provides an initial attempt at installing .pyi files for #3558, but I am not convinced that the installation is to the correct location for the wheel builds.

**Review by dzenanz — APPROVED** [review]

**Pull request merged**